iceoryx_posh/source/roudi/application/roudi_app.cpp

```diff
@@ -39,11 +40,22 @@ namespace roudi
 // using unnamed namespace to keep the functions in this translation unit
 namespace
 {
-sem_t g_runSemaphore;
+static posix::Semaphore g_runSemaphore{[] {
+    auto runSemaphore = posix::Semaphore::create(ROUDI_SEMAPHORE_NAME, O_CREAT | O_EXCL, 0);
+    if (runSemaphore.has_error())
+    {
+        std::cerr << "Unable to create runSemaphore \"" << ROUDI_SEMAPHORE_NAME
+                  << "\", either RouDi is already running or someone else has already created this semaphore."
+                  << std::endl;
+        std::terminate();
```

Comment:
Why not calling the error handler with a FATAL error? We need this source of information

Reply:
Done